app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/choose/ChooseContractForCertificate.kt

```diff
@@ -0,0 +1,156 @@
+import com.hedvig.android.data.travelcertificate.ContractEligibleWithAddress
+import com.hedvig.android.feature.travelcertificate.ui.FullScreenLoading
+import hedvig.resources.R
+
+@Composable
+internal fun ChooseContractForCertificateDestination(
+  viewModel: ChooseContractForCertificateViewModel,
+  navigateBack: () -> Unit,
+  onContinue: (String) -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  ChooseContractForCertificate(
+    uiState = uiState,
+    navigateBack = navigateBack,
+    onContinue = onContinue,
+    reload = { viewModel.emit(ChooseContractEvent.RetryLoadData) },
+  )
+}
+
+@Composable
+private fun ChooseContractForCertificate(
+  uiState: ChooseContractUiState,
+  navigateBack: () -> Unit,
+  onContinue: (String) -> Unit,
+  reload: () -> Unit,
+) {
+  var selectedContractId by remember {
+    mutableStateOf<String?>(null)
+  }
+  when (uiState) {
+    ChooseContractUiState.Failure -> {
+      ShowFailure(
+        navigateBack = navigateBack,
+        reload = reload,
+      )
+    }
+    ChooseContractUiState.Loading -> {
+      FullScreenLoading()
+    }
+    is ChooseContractUiState.Success -> {
+      HedvigScaffold(
+        navigateUp = navigateBack,
+        modifier = Modifier.clearFocusOnTap(),
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        Text(
+          text = stringResource(id = R.string.travel_certificate_select_contract_title),
+          style = MaterialTheme.typography.headlineSmall,
+          textAlign = TextAlign.Center,
+          modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        )
+        Spacer(Modifier.weight(2f))
+        for (contract in uiState.eligibleContracts) {
+          HedvigCard(
+            onClick = { selectedContractId = contract.contractId },
+            modifier = Modifier
+              .fillMaxWidth()
+              .padding(horizontal = 16.dp),
+          ) {
+            Row(
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier
+                .heightIn(72.dp)
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp, vertical = 10.dp),
+            ) {
+              Text(
+                text = contract.address,
+                style = MaterialTheme.typography.headlineSmall,
+                modifier = Modifier.weight(1f),
+              )
+              Spacer(Modifier.width(8.dp))
+              SelectIndicationCircle(selectedContractId == contract.contractId)
+            }
+          }
+          Spacer(modifier = (Modifier.height(4.dp)))
+        }
+        Spacer(modifier = Modifier.height(12.dp))
+        HedvigContainedButton(
+          onClick = { selectedContractId?.let { onContinue(it) } },
+          modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        ) {
+          Text(text = stringResource(id = R.string.general_continue_button))
```

Comment:
```suggestion
          Text(
            text = stringResource(id = R.string.general_continue_button),
            style = MaterialTheme.typography.headlineSmall
          )
```
And this is this probably